### PR TITLE
feat: [IDP-943]: Harness Data Source checks data values fetching

### PR DESCRIPTION
**Pull request by devesh-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51811)

### CODE REVIEW

**Review by vikyathharekal — COMMENTED** [review]
> Reviewed 19 of 37 files at r1, 28 of 28 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51811)* status: Some of the checks are too old. 47 file(s) to review. 0 of 2 approvals obtained. 

___
*[`idp-service/src/main/java/io/harness/idp/scorecard/datapoints/constants/DataPoints.java` line 24 at r2](https://reviewable.io/reviews/harness/harness-core/51811#-NcRS1mo-IEfSzW1ex51:-NcRS1mo-IEfSzW1ex52:b-22pxnc) ([raw file](https://github.com/harness/harness-core/blob/4a9f485edcea388561151f4ebc6cc5def9fc62dd/idp-service/src/main/java/io/harness/idp/scorecard/datapoints/constants/DataPoints.java#L24)):*
> ```Java
>   public static final String IS_POLICY_EVALUATION_SUCCESSFUL_IN_PIPELINE = "isPolicyEvaluationSuccessful";
>   public static final String PERCENTAGE_OF_CI_PIPELINE_FAILING_IN_SEVEN_DAYS = "%ofCIPipelinePassingInPast7Days";
>   public static final String PIPELINE_TEST_FAILING_IN_CI_IS_ZERO = "testFailingOnCiPipelineIsZero";
> ```

How about `noTestsFailingInCiPipeline`?



_Code quote:_
```Java
testFailingOnCiPipelineIsZero
```
___
*[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/factory/PipelineExecutionResponseFactory.java` line 9 at r2](https://reviewable.io/reviews/harness/harness-core/51811#-NcRTCC1Dpskk-T9rO6C:-NcRTCC1Dpskk-T9rO6D:b-t07dj6) ([raw file](https://github.com/harness/harness-core/blob/4a9f485edcea388561151f4ebc6cc5def9fc62dd/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/factory/PipelineExecutionResponseFactory.java#L9)):*
> ```Java
> package io.harness.idp.scorecard.datapointsdata.datapointvalueparser.factory;
> 
> import static io.harness.idp.scorecard.datapoints.constants.DataPoints.*;
> ```

nit:  we could avoid wildcard imports



_Code quote:_
```Java
*
```
___
*[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/impl/PipelinePolicyEvaluationParser.java` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/51811#-NcRTfWm9hCUDRTVC3KZ:-NcRTfWm9hCUDRTVC3K_:b-vtv9sw) ([raw file](https://github.com/harness/harness-core/blob/4a9f485edcea388561151f4ebc6cc5def9fc62dd/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/impl/PipelinePolicyEvaluationParser.java#L25)):*
> ```Java
>     Map<String, Object> returnData = new HashMap<>();
> 
>     if (responseCI != null) {
> ```

can we move this to a common method and pass in CI and CD response to get back the boolean?
___
*[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/dsldataprovider/DslConstants.java` line 22 at r2](https://reviewable.io/reviews/harness/harness-core/51811#-NcRU9c93AVIMnL2_SR8:-NcRU9c93AVIMnL2_SR9:bkdpqs4) ([raw file](https://github.com/harness/harness-core/blob/4a9f485edcea388561151f4ebc6cc5def9fc62dd/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/dsldataprovider/DslConstants.java#L22)):*
> ```Java
>   public static final String CD_SERVICE_IDENTIFIER_KEY = "cdServiceIdentifier";
> 
>   public static final long ThirtyDaysInMillis = 30 * 24 * 60 * 60 * 1000;
> ```

Can we have these 2 constants in ALL_CAPS with _ in between?



_Code quote:_
```Java
ThirtyDaysInMillis
```
___
*[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/dsldataprovider/DslConstants.java` line 22 at r2](https://reviewable.io/reviews/harness/harness-core/51811#-NcRUVpkAsW03hR2cMMj:-NcRUVpkAsW03hR2cMMk:bsswkeo) ([raw file](https://github.com/harness/harness-core/blob/4a9f485edcea388561151f4ebc6cc5def9fc62dd/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/dsldataprovider/DslConstants.java#L22)):*
> ```Java
>   public static final String CD_SERVICE_IDENTIFIER_KEY = "cdServiceIdentifier";
> 
>   public static final long ThirtyDaysInMillis = 30 * 24 * 60 * 60 * 1000;
> ```

This could lead to an integer overflow. We could have one of the numbers here as long. For ex - 30L * 24 * 60 * 60 * 1000



_Code quote:_
```Java
30 
```

**Comment by devesh-harness** [comment]
trigger uts

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/tiserviceclient/TIServiceClient.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
```
Comment (raghavharness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/tiserviceclient/TIServiceClient.java:9]: Can we move this package to a common module ? Maybe in `879-pipeline-ci-commons/src/main/java/io/harness/ci`. Therefore you won't need to import the whole `//332-ci-manager/service:module` .
cc: @harshjain12

**Review by mankrit-singh — APPROVED** [review]

**Review by raghavharness — DISMISSED** [review]
CI changes looks good.

**Comment by devesh-harness** [comment]
trigger uts

**Review by vikyathharekal — DISMISSED** [review]
> Reviewed 11 of 12 files at r3, 7 of 7 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51811)* status: 47 file(s) to review. 1 of 2 approvals obtained.

**Comment by devesh-harness** [comment]
trigger ut1

**Comment by devesh-harness** [comment]
trigger ut1

**Comment by devesh-harness** [comment]
trigger ut1

**Review by raghavharness — DISMISSED** [review]

**Review by mankrit-singh — DISMISSED** [review]

**Review by pranay-harness — DISMISSED** [review]
Place holder secret

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut7

**Comment by devesh-harness** [comment]
trigger ut0, ut1, ut2, ut3, ut4

**Comment by devesh-harness** [comment]
trigger ut0, ut1, ut2, ut3, ut4

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut0

**Comment by harshjain12** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger uts

**Review by pranay-harness — APPROVED** [review]

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51811)* status: 45 file(s) to review.

**Review by raghavharness — APPROVED** [review]

**Comment by devesh-harness** [comment]
trigger review

**Review by mankrit-singh — APPROVED** [review]

**Pull request merged**